lib/LaTeXML/Engine/TeX_Math.pool.ltxml

```diff
@@ -670,53 +692,67 @@ DefConstructor('\mathinner{}', "?#1(<ltx:XMWrap role='ATOM'>#1</ltx:XMWrap>)()",
-  '\|'         => { char => "\x{2225}", lrole => 'VERTBAR', rrole => 'VERTBAR' },
-  '\uparrow'   => { char => "\x{2191}", lrole => 'OPEN', rrole => 'CLOSE', name => 'uparrow' },   # ??
-  '\Uparrow'   => { char => "\x{21D1}", lrole => 'OPEN', rrole => 'CLOSE', name => 'Uparrow' },   # ??
-  '\downarrow' => { char => "\x{2193}", lrole => 'OPEN', rrole => 'CLOSE', name => 'downarrow' }, # ??
-  '\Downarrow' => { char => "\x{21D3}", lrole => 'OPEN', rrole => 'CLOSE', name => 'Downarrow' }, # ??
-  '\updownarrow' => { char => "\x{2195}", lrole => 'OPEN', rrole => 'CLOSE', name => 'updownarrow' }, # ??
-  '\Updownarrow' => { char => "\x{21D5}", lrole => 'OPEN', rrole => 'CLOSE', name => 'Updownarrow' }, # ??
+  ('(' => { lrole => 'OPEN', rrole => 'CLOSE' },
+  ')'        => { lrole => 'OPEN', rrole => 'CLOSE' },
+  '['        => { lrole => 'OPEN', rrole => 'CLOSE' },
+  ']'        => { lrole => 'OPEN', rrole => 'CLOSE' },
+  '{'        => { lrole => 'OPEN', rrole => 'CLOSE' },
+  '}'        => { lrole => 'OPEN', rrole => 'CLOSE' },
+  "\x{230A}" => { lrole => 'OPEN', rrole => 'CLOSE', name => 'lfloor' },
+  "\x{230B}" => { lrole => 'OPEN', rrole => 'CLOSE', name => 'rfloor' },
+  "\x{2308}" => { lrole => 'OPEN', rrole => 'CLOSE', name => 'lceil' },
+  "\x{2309}" => { lrole => 'OPEN', rrole => 'CLOSE', name => 'rceil' },
+  "\x{27E8}" => { lrole => 'OPEN', rrole => 'CLOSE', name => 'langle' },
+  "\x{27E9}" => { lrole => 'OPEN', rrole => 'CLOSE', name => 'rangle' },
+  '<'        => { lrole => 'OPEN', rrole => 'CLOSE', name => 'langle',
+    char => "\x{27E8}", meaning => undef },
+  '>' => { lrole => 'OPEN', rrole => 'CLOSE', name => 'rangle',
+    char => "\x{27E9}", meaning => undef },
+  '/'        => { lrole => 'MULOP',   rrole => 'MULOP' },
+  UTF(0x5C)  => { lrole => 'MULOP',   rrole => 'MULOP', name => 'backslash' },
+  '|'        => { lrole => 'VERTBAR', rrole => 'VERTBAR' },
+  "\x{2225}" => { lrole => 'VERTBAR', rrole => 'VERTBAR' },
+  "\x{2191}" => { lrole => 'OPEN',    rrole => 'CLOSE', name => 'uparrow' },        # ??
+  "\x{21D1}" => { lrole => 'OPEN',    rrole => 'CLOSE', name => 'Uparrow' },        # ??
+  "\x{2193}" => { lrole => 'OPEN',    rrole => 'CLOSE', name => 'downarrow' },      # ??
+  "\x{21D3}" => { lrole => 'OPEN',    rrole => 'CLOSE', name => 'Downarrow' },      # ??
+  "\x{2195}" => { lrole => 'OPEN',    rrole => 'CLOSE', name => 'updownarrow' },    # ??
+  "\x{21D5}" => { lrole => 'OPEN',    rrole => 'CLOSE', name => 'Updownarrow' },    # ??
   );
 
-# With new treatment of Simple Symbols as just Box's with assigned attributes,
-# we're not getting whatsits, and so we're not looking them up the same way!!!
-# TEMPORARILY (?) hack the Delimiter map
-foreach my $entry (values %DELIMITER_MAP) {
-  $DELIMITER_MAP{ $$entry{char} } = $entry; }
+# The \lx@left, \lx@right versions are like \left,\right but without any extra grouping
+DefConstructor('\lx@left TeXDelimiter',
+  "?#hint(<ltx:XMHint/>)(#1)",
```

Comment:
What is the idea behind the empty XMHint elements? I see they are used in a few definitions here. I wonder if they should carry more information to be useful - or are they already useful when empty?

I see the code adds them when the `\left`/`\right` is applied to a dot, which indeed has no visible trace in rendering. Maybe an attribute on XMHint could indicate some of that?

Reply:
I inherited the hint from the previous code, but I doubt the hint is ever actually used. Probably put there either so there's at least 1 element, or "just in case".